### DISCUSSION TITLE
Error in ftINIT(line 285)

### DISCUSSION THREAD

**Discussion by Mars-hyj** [SysBioChalmers/Human-GEM#521] [Q&A]
Hi, I followed the Human-GEM-guide to run ftINIT and it came up with the following error: 

`model1 = ftINIT(prepData, data_struct.tissues{1}, [], [], data_struct, {}, getHumanGEMINITSteps('1+0'), false, true);`
`The function or variable 'rxnsTurnedOn1' is not recognized.
Error in ftINIT (line 285)
    rxnsTurnedOn = rxnsTurnedOn | rxnsTurnedOn1.'.`

And the reference model fails all of the essential metabolic tasks.
`>> checkTasks(ihuman, [], true, false, false, essentialTasks);
WARNING: Exchange metabolites should normally not be removed from the model when using checkTasks. Inputs and outputs are defined in the task file instead. Use importModel(file,false) to import a model with exchange metabolites remaining

FAIL: [ER] Aerobic rephosphorylation of ATP from glucose
FAIL: [ER] Aerobic rephosphorylation of GTP
FAIL: [ER] Aerobic rephosphorylation of CTP
FAIL: [ER] Aerobic rephosphorylation of UTP
FAIL: [BS] ATP de novo synthesis
FAIL: [BS] CTP de novo synthesis
FAIL: [BS] GTP de novo synthesis
FAIL: [BS] UTP de novo synthesis
FAIL: [BS] dATP de novo synthesis
FAIL: [BS] dCTP de novo synthesis
FAIL: [BS] dGTP de novo synthesis
FAIL: [BS] dTTP de novo synthesis
FAIL: [SU] Histidine uptake
FAIL: [SU] Isoleucine uptake
FAIL: [SU] Leucine uptake
FAIL: [SU] Lysine uptake
FAIL: [SU] Methionine uptake
FAIL: [SU] Phenylalanine uptake
FAIL: [SU] Threonine uptake
FAIL: [SU] Tryptophan uptake
FAIL: [SU] Valine uptake
FAIL: [IC] Glycerate 3-phosphate  de novo synthesis
FAIL: [IC] Mitochondrial acetyl-CoA de novo synthesis
FAIL: [IC] Mitochondrial AKG de novo synthesis
FAIL: [IC] Erythrose 4-phosphate de novo synthesis
FAIL: [IC] Fructose 6-phosphate de novo synthesis
FAIL: [IC] Glyceraldehyde 3-phosphate de novo synthesis
FAIL: [IC] Glucose 6-phosphate de novo synthesis
FAIL: [IC] Mitochondrial oxaloacetate de novo synthesis
FAIL: [IC] Phosphoenolpyruvate de novo synthesis
FAIL: [IC] Pyruvate de novo synthesis
FAIL: [IC] Ribose 5-phosphate de novo synthesis
FAIL: [IC] Mitochondrial succinyl-CoA de novo synthesis
FAIL: [BS] Cholesterol de novo synthesis
FAIL: [BS] Protein synthesis from AAs
FAIL: [ER] Oxidative phosphorylation
FAIL: [ER] Oxidative decarboxylation
FAIL: [ER] Krebs cycle NADH
FAIL: [ER] Ubiquinol-to-proton
FAIL: [ER] Ubiquinol-to-ATP
FAIL: [SU] Beta oxidation of saturated FA
FAIL: [SU] Beta oxidation of long-chain FA
FAIL: [SU] Beta oxidation of odd-chain FA
FAIL: [SU] Beta oxidation of unsaturated fatty acid (n-9) 
FAIL: [SU] Beta oxidation of unsaturated fatty acid (n-6) 
FAIL: [SU] Uptake and beta oxidation of all NEFAs
FAIL: [SU] Choline uptake
FAIL: [SU] Inositol uptake
FAIL: [BS] Phosphatidylcholine de novo synthesis
FAIL: [BS] Phosphatidylethanolamine de novo synthesis
FAIL: [BS] Phosphatidylserine de novo synthesis
FAIL: [BS] Phosphatidylinositol de novo synthesis
FAIL: [BS] Thiamin phosphorylation to TPP
FAIL: [BS] Coenzyme A synthesis from pantothenate
FAIL: [BS] FAD synthesis from riboflavin
FAIL: [BS] Heme biosynthesis
FAIL: [GR] Growth on Ham's media (biomass production)`

What should I do to troubleshoot the problem? Any help would be really appreciated. By the way, I would like to know where the available preData is. 

> For version 1.12.0 of Human-GEM, the prepData is available in the Zenodo .zip file. (ftINIT guide)

**Comment by mihai-sysbio** [comment]
Hi @Mars-hyj, thanks for giving Human-GEM and ftINIT a try!

Typically, function names that normally should be recognized end up being not recognized if the folder that contains them is not added to the Matlab path. Are you running a single version of RAVEN, and if so, which one?

Regarding the data from Zenodo, here is the right paragraph https://sysbiochalmers.github.io/Human-GEM-guide/gem_extraction/#retrieve-the-data
> To demonstrate the use of ftINIT, we will walk through an example where we generate models from GTEx data. For simplicity, we extracted 60 RNA-Seq profiles from in total 12 tissues. The data can be downloaded [here](https://doi.org/10.5281/zenodo.6811073) together with other data useful for this tutorial.

**Reply by Mars-hyj** [reply]
Hi, thanks!

I am using RAVEN 2.7.6 and have added it to the Matlab path. I'm looking for the preData.mat mentioned in this paragraph

> ftINIT has a preparation step that needs to be run once for a reference model such as Human-GEM. The purpose of the preparation step is to make calculations in advance to reduce the time it takes to generate each context-specific model. The result is a prepData structure that contains various information needed by ftINIT. This operation takes 1-2 hours on a standard laptop computer, so make sure to save it once it has completed. For version 1.12.0 of Human-GEM, the prepData is available in the Zenodo .zip file.

**Reply by mihai-sysbio** [reply]
@Mars-hyj I'm unsure what the question is; `prepData.mat` is the file obtained with the code just below the one you copied, also included here for reference:

> % The second flag indicates if the model should be converted to gene symbols from ENSEMBL. This has to be decided at this point.
% Replace path/to/HumanGEM with your local path to the Human-GEM repo root.
% For use with animal models derived from Human-GEM, such as Mouse-GEM, both the model and paths needs to be replaced. Also, 
% the convert genes flag may be irrelevant depending on the if ENSEMBL genes are used in that model.
prepData = prepHumanModelForftINIT(ihuman, false, 'path/to/HumanGEM/data/metabolicTasks/metabolicTasks_Essential.txt', 'path/to/HumanGEM/model/reactions.tsv');
save('**prepData.mat'**, 'prepData')
